onnxruntime/core/graph/graph.cc

```diff
@@ -666,12 +666,17 @@ void Node::ToProto(NodeProto& proto, bool update_subgraphs) const {
 
   // Set attributes.
   proto.clear_attribute();
-  for (const auto& attribute : attributes_) {
+  for (const auto& [name, attribute] : attributes_) {
     const gsl::not_null<AttributeProto*> attr{proto.add_attribute()};
-    *attr = attribute.second;  // copy
-    if (update_subgraphs && attr->has_g()) {
+    *attr = attribute;  // copy
+    if (update_subgraphs && utils::HasGraph(*attr)) {
+      auto find_hit = attr_to_subgraph_map_.find(name);
+      ORT_ENFORCE(find_hit != attr_to_subgraph_map_.cend(),
+                  "Node [", name_, "] op_type [", op_type_, "] does not have the subgraph for ", name);
```

Comment:
nit: ORT_ENFORCE adds binary size. what use case makes this check required? I would have expected validation when the Node is created to be sufficient.